src/main/java/de/tum/in/www1/artemis/service/scheduled/ProgrammingExerciseScheduleService.java

```diff
@@ -79,17 +89,18 @@ public void scheduleRunningExercisesOnStartup() {
             }
             SecurityUtils.setAuthorizationObject();
 
-            List<ProgrammingExercise> programmingExercisesWithBuildAfterDueDate = programmingExerciseRepository
-                    .findAllByBuildAndTestStudentSubmissionsAfterDueDateAfterDate(ZonedDateTime.now());
-            programmingExercisesWithBuildAfterDueDate.forEach(this::scheduleExercise);
+            List<ProgrammingExercise> programmingExercisesWithDueDateOrBuildAfterDueDate = programmingExerciseRepository
```

Comment:
I was really confused by the name and it took me a while to understand it
To make it easier, I suggest we rename this to something like `programmingExercisesWithFutureDueDateOrBuildAfterDueDate`

Reply:
Got resolved by the change described in the other comment.